src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  return sourceString
+    .split(';')
+    .filter((line) => line.trim())
+    .reduce((prev, item) => {
+      const [key, value] = item.split(':').map((part) => part && part.trim());
 
+      if (key && value) {
+        return { ...prev, [key]: value };
+      }
```

Comment:
There is a missing return statement in the `reduce` function. If the condition `if (key && value)` is true, the function should return the updated accumulator. Otherwise, it will return `undefined` for each iteration, causing the final result to be incorrect.